mux/README.md

```diff
@@ -0,0 +1,41 @@
+# Mux
+
+`mux` package provides protocol multiplexing for HTTP & gRPC servers.
+
+## Usage
+
+```golang
+package main
+
+import (
+	"net/http"
+
+	"github.com/odpf/salt/mux"
+)
+
+func main() {
+	ctx := context.Background()
+	addr := ":8080"
+
+	grpcServer, err := mux.NewGRPCServer()
+	if err != nil {
+		panic(err)
+	}
+
+	grpcGateway, err := grpcServer.Gateway(addr)
```

Comment:
Let's add an example of registering the common version grpc service here? ref - https://github.com/odpf/salt/blob/main/common/service.go